examples/todo/src/tests.rs

```diff
@@ -1,25 +1,37 @@
-extern crate parking_lot;
 extern crate rand;
 
+use std::collections::HashMap;
+
 use super::task::Task;
-use self::parking_lot::Mutex;
 use self::rand::{Rng, thread_rng, distributions::Alphanumeric};
 
+use rocket::config::{Config, Environment};
 use rocket::local::Client;
 use rocket::http::{Status, ContentType};
 
-// We use a lock to synchronize between tests so DB operations don't collide.
-// For now. In the future, we'll have a nice way to run each test in a DB
-// transaction so we can regain concurrency.
-static DB_LOCK: Mutex<()> = Mutex::new(());
-
 macro_rules! run_test {
     (|$client:ident, $conn:ident| $block:expr) => ({
-        let _lock = DB_LOCK.lock();
-        let (rocket, db) = super::rocket();
+        // Generate a random database path for each test
+        let mut rng = thread_rng();
+        let db_name: String = rng.sample_iter(&Alphanumeric).take(6).collect();
+        let db_path = std::env::temp_dir().join(format!("rocket_todo_{}.db", db_name));
+
+        let mut sqlite_database = HashMap::new();
+        sqlite_database.insert("url", db_path.to_str().expect("path is valid UTF-8").to_string());
+        let mut databases = HashMap::new();
+        databases.insert("sqlite_database", sqlite_database);
+
+        let config = Config::build(Environment::Development)
+            .extra("databases", databases)
+            .expect("valid config");
+
+        let (rocket, db) = super::rocket(Some(config));
+
         let $client = Client::new(rocket).expect("Rocket client");
         let $conn = db.expect("failed to get database connection for testing");
         $block
+
+        let _ = std::fs::remove_file(db_path);
```

Comment:
If the test fails, won't this file stick around forever? Can we push this line up and have the removal happen on unwind?

Reply:
"forever" in this case is until the temporary directory is cleared, but it's definitely not ideal. Now that you mention it, I'm afraid the file removal will unconditionally fail on windows because a connection is still open by this point. I'll have to think about this a bit more, maybe a `Drop` guard is needed to solve this.